python/ray/util/sgd/torch/examples/segmentation/train_segmentation.py

```diff
@@ -0,0 +1,266 @@
+from torch import nn
+import torchvision
+
+import ray
+from ray.util.sgd.torch.examples.segmentation.coco_utils import get_coco
+import ray.util.sgd.torch.examples.segmentation.transforms as T
+import ray.util.sgd.torch.examples.segmentation.utils as utils
+from ray.util.sgd.torch import TrainingOperator
+from ray.util.sgd import TorchTrainer
+
+
+def get_dataset(name,
+                image_set,
+                transform,
+                num_classes_only=False,
+                download="auto"):
+    def sbd(*args, **kwargs):
+        return torchvision.datasets.SBDataset(
+            *args, mode="segmentation", **kwargs)
+
+    paths = {
+        "voc": (os.path.expanduser("~/datasets01/VOC/060817/"),
+                torchvision.datasets.VOCSegmentation, 21),
+        "voc_aug": (os.path.expanduser("~/datasets01/SBDD/072318/"), sbd, 21),
+        "coco": (os.path.expanduser("~/datasets01/COCO/022719/"), get_coco, 21)
+    }
+    p, ds_fn, num_classes = paths[name]
+    if num_classes_only:
+        return None, num_classes
+
+    if download == "auto" and os.path.exists(p):
+        download = False
+
+    ds = ds_fn(p, download=download, image_set=image_set, transforms=transform)
+    return ds, num_classes
+
+
+def data_creator(config):
+    dataset, num_classes = get_dataset(
+        args.dataset, "train", get_transform(train=True))
+    config["num_classes"] = num_classes
+    dataset_test, _ = get_dataset(
+        args.dataset, "val", get_transform(train=False))
+    data_loader = torch.utils.data.DataLoader(
+        dataset,
+        batch_size=args.batch_size,
+        num_workers=args.data_workers,
+        collate_fn=utils.collate_fn,
+        drop_last=True)
+
+    data_loader_test = torch.utils.data.DataLoader(
+        dataset_test,
+        batch_size=1,
+        num_workers=args.data_workers,
+        collate_fn=utils.collate_fn)
+    return data_loader, data_loader_test
+
+
+def get_transform(train):
+    base_size = 520
+    crop_size = 480
+
+    min_size = int((0.5 if train else 1.0) * base_size)
+    max_size = int((2.0 if train else 1.0) * base_size)
+    transforms = []
+    transforms.append(T.RandomResize(min_size, max_size))
+    if train:
+        transforms.append(T.RandomHorizontalFlip(0.5))
+        transforms.append(T.RandomCrop(crop_size))
+    transforms.append(T.ToTensor())
+    transforms.append(
+        T.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225]))
+
+    return T.Compose(transforms)
+
+
+def criterion(inputs, target):
+    losses = {}
+    for name, x in inputs.items():
+        losses[name] = nn.functional.cross_entropy(x, target, ignore_index=255)
+
+    if len(losses) == 1:
+        return losses["out"]
+
+    return losses["out"] + 0.5 * losses["aux"]
+
+
+class SegOperator(TrainingOperator):
+    def setup(self, config):
+        pass
+        # Currently, this LR scheduler is broken due to
+        # https://github.com/pytorch/pytorch/issues/32639. But theoretically,
```

Comment:
Not sure why exactly this issue applies. Could you explain it more? Especially since it keeps getting closed by the community bot due to being stale, so it's extra confusing that we say we are blocked on a closed issue.